sixtyfps_runtime/interpreter/dynamic_component.rs

```diff
@@ -300,6 +300,26 @@ pub struct ComponentDescription<'id> {
 
     /// compiled globals
     compiled_globals: Vec<crate::global_component::CompiledGlobal>,
+    /// Map of all exported global singletons and their public properties/callbacks
+    exported_globals: BTreeMap<String, Rc<BTreeMap<String, PropertyDeclaration>>>,
```

Comment:
This information should be accessible from the compiled globals. Is that not working?

Reply:
Hmm, the issue is that accessing the original requires one of those guards, which breaks the impl Iterator + ‘_. We did the same with the public properties. But I could probably change this to a map to a vec index and store the public props in the compiled globals entries. I’ll try that, might look better.

Reply:
I've done that now in a follow-up commit. It adds a bit more code overall, but it improves the locality (in terms of readability of the code).